### PR TITLE
Add docs re stable branch

### PR DESCRIPTION
**Pull request by max-sixty**
ref https://github.com/pydata/xarray/pull/4440#issuecomment-695836637

Not sure this is the best approach; very open minded to alternatives

### CODE REVIEW

File: HOW_TO_RELEASE.md [hunk 1]
```diff
@@ -15,8 +15,14 @@ upstream        https://github.com/pydata/xarray (push)
 
  1. Ensure your master branch is synced to upstream:
      ```sh
+     git switch master
```
Comment (keewis) [inline HOW_TO_RELEASE.md:18]: is this a new feature? The version of git I'm using only has `checkout`

File: HOW_TO_RELEASE.md [hunk 2]
```diff
@@ -15,8 +15,14 @@ upstream        https://github.com/pydata/xarray (push)
...
      git pull upstream master
      ```
+ 2. Confirm there are no commits on stable that are not yet merged
+    ([ref](https://github.com/pydata/xarray/issues/2821)):
```
Comment (keewis) [inline HOW_TO_RELEASE.md:22]:
```suggestion
    ([ref](https://github.com/pydata/xarray/pull/4440)):
```
?

File: HOW_TO_RELEASE.md (hunk 1 above)
Comment (max-sixty) [reply HOW_TO_RELEASE.md:18]: Looks like it's 2.23, last August! Too new?

Comment (keewis) [reply HOW_TO_RELEASE.md:18]: yeah, well, I've got 2.20.1, which is the version included in debian stable (released in July 2019). Nothing too serious, I guess, maintainers should know how to switch branches.

Comment (max-sixty) [reply HOW_TO_RELEASE.md:18]: OK!

**Pull request merged**